src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/TypeHelpers.cs

```diff
@@ -439,9 +434,21 @@ internal static DkmCustomUIVisualizerInfo[] GetDebuggerCustomUIVisualizerInfo(th
             return result;
         }
 
-        private static string EvaluateDebuggerDisplayString(DkmClrValue value, string str)
+        internal static void EvaluateDebuggerDisplayStringAndContinue(this DkmClrValue value, DkmWorkList workList, DkmInspectionContext inspectionContext, DkmClrType targetType, string str, DkmCompletionRoutine<DkmEvaluateDebuggerDisplayStringAsyncResult> completionRoutine)
+        {
+            if (str == null)
+            {
+                completionRoutine(default(DkmEvaluateDebuggerDisplayStringAsyncResult));
+            }
+            else
+            {
+                value.EvaluateDebuggerDisplayString(workList, inspectionContext, targetType, str, completionRoutine);
+            }
+        }
+
+        internal static string GetResult(this DkmEvaluateDebuggerDisplayStringAsyncResult asyncResult)
         {
-            return str == null ? null : value.EvaluateDebuggerDisplayString(str);
+            return !asyncResult.Equals(default(DkmEvaluateDebuggerDisplayStringAsyncResult)) ? asyncResult.Result : null;
```

Comment:
Comparing value types is relatively expensive. Consider using asyncResult.Result in the caller and removing this method.